### ISSUE TITLE
[$1000] [CRITICAL] Backwards Compatibility: Ensure rejected reports on OldDot translate to NewDot

### ISSUE THREAD

**Issue by m-natarajan** [Expensify/App#32243]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.5-3
**Reproducible in staging?:** y
**Reproducible in production?:** y
**If this was caught during regression testing, add the test name, ID and link from TestRail:**
**Email or phone of affected tester (no customers):**
**Logs:** https://stackoverflow.com/c/expensify/questions/4856
**Expensify/Expensify Issue URL:**
**Issue reported by:** @JmillsExpensify 
**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1701212831061939

## Action Performed:

Note: Account A is Approver, Account B is Requestor

OldExpensify

1. Sign in to Account A at www.expensify.com
2. Go to Settings > Workspaces > Group > New Workspace > choose Collect plan (the Collect Plan is important)
3. Go to Settings > Workspaces > Group > [Workspace Name] > Members > invite Account B

NewExpensify 

1. Sign in to Account B at https://staging.new.expensify.com/ 
4. Click the Green '+' > Request Money > Manual > any amount > request from Account A
5. Go to chat with Account A  > click the '+' 
6. Request Money at least 3 more times (creating multiple requests) 

OldExpensify

1. Sign in as Account B at www.expensify.com
2. Click on the Expenses page
3. Click on Reports > click on the report to see the expenses
4. Click Undo Submit (the report will become Open)
5. Click back on Expenses > click the checkbox to the left of the expense > Click Add to Report > New
6. Click Report > find the new report and click on it to see the expense
7. Click Details of the and move it to the Collect Plan created by Account A
8. Submit the Report to Account A

OldExpensify

1. Sign in as Account A at www.expensify.com
2. Find the Submitted report
3. Click on one of the expenses to view the details
4. Click Reject

## Expected Result: 
The request is removed from the report on NewDot

## Actual Result: 
The request remains on the report in NewDot

## Workaround:
unknown
## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

![2023-11-30_14-20-36 (7)](https://github.com/Expensify/App/assets/51066321/f7f14429-6cd5-4a70-9d3f-13e7e06ac5e5)

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01df455b9f7da1802b</li>
        <li>Upwork Job ID: 1730025610785619968</li>
        <li>Last Price Increase: 2023-12-08</li>
    </ul>
</details>

**Comment by junsupark94** [comment]
Could you show how you created the report and its multiple requests in NewDot?

I created a chat and made requests but it doesn't show up in Reports section of OldDot. 
I created a workspace and made requests and it does show up in Reports but there is no delete button and the status says Processing instead of Open.

**Comment by Christinadobrzyn** [comment]
We're still waiting on proposals for this.

**Comment by keisyrzk** [comment]
Still not able to log into old expensify. I am redirected to the new dot.

**Comment by junsupark94** [comment]
@Christinadobrzyn There are some issues accessing OldDot with my email address with +numbers appended.
- I log out of NewDot. Go to OldDot which takes me to NewDot signed in.
- I open Incognito mode and go to OldDot. I sign in, enter the code, and it takes me to newDot.

I was able to access OldDot using incognito mode and a completely different email address. I'll try to reproduce the bug and report back.

**Comment by Christinadobrzyn** [comment]
Thanks @junsupark94! Please keep us in the loop!

**Comment by AmjedNazzal** [comment]
Just to clarify, when the "Actual result" says "The request remains on the report in NewDot", is that talking about the money request showing in NewDot on account B in the report between account A and account B like shown in the below screenshot? note that the screenshot is taken from account B 

<img width="1280" alt="bugimg" src="https://github.com/Expensify/App/assets/74202751/3a6c88ee-aaef-4c2b-a403-cb415c6ef057">

**Comment by nikosamofa** [comment]
@m-natarajan Can you record the workflow again? I cannot follow the recording.
Especially, 

> 2. Click on the Expenses page
> 3. Click on Reports > click on the report to see the expenses
> 4. Click Undo Submit (the report will become Open)

I cannot find the button `Undo Submit`

**Comment by shubham1206agra** [comment]
@Christinadobrzyn, I am confident that this is a backend issue. This also broke my 1:1 DM with account B (in Account A).
Can you make this internal? I think this is an unimplemented feature in NewDot.

**Comment by Christinadobrzyn** [comment]
Thanks @shubham1206agra - making this internal for our engineers to work on along with @eVoloshchak.

**Comment by Christinadobrzyn** [comment]
@NikkiWines let me know what you need here - in my testing the steps in the OP help to reproduce but let me know if you need anything else!

**Comment by Christinadobrzyn** [comment]
Awesome! Thanks @NikkiWines! I'm going to move this to the Wave 6 project for tracking.

**Comment by greg-schroeder** [comment]
I'm going to add this under Critical as w/o true backwards compatibility it's honestly not particularly viable to ask a real customer to try these flows

**Comment by greg-schroeder** [comment]
Holding for https://github.com/Expensify/Expensify/issues/343041 as this may get resolved naturally

**Comment by Christinadobrzyn** [comment]
looks like https://github.com/Expensify/Expensify/issues/343041 is done - i'll try to test this again tomorrow

**Comment by Christinadobrzyn** [comment]
I keep getting an error before I start this process, @eVoloshchak are you able to reproduce the issue reported in the OP?

**Comment by Christinadobrzyn** [comment]
I'm going ooo next week so I'm assigning a BZ teammate to monitor this while I'm gone

Status- trying to retest the steps in the OP to see if it's still happening and we should review proposals.

**Comment by sakluger** [comment]
I eemoved HOLD from the issue title since the hold issue was deployed. I also asked for help retesting in the QA Slack channel: https://expensify.slack.com/archives/C9YU7BX5M/p1704909880155769.

**Comment by kbecciv** [comment]
Issue is still reproducible on build 1.4.24.0


https://github.com/Expensify/App/assets/93399543/a32f74fc-3181-4430-ad52-3fc091400a2f

**Comment by sakluger** [comment]
@greg-schroeder @NikkiWines it seems that https://github.com/Expensify/Expensify/issues/343041 did not fix this issue. Are there any other backend compatibility projects in the works that we should put this on hold for? If not, is there an engineer from Wave6 that can work on it?

**Comment by NikkiWines** [comment]
There's a still a [PR](https://github.com/Expensify/Web-Expensify/pull/40051) in the works for https://github.com/Expensify/Expensify/issues/343041 which is probably why this is still occurring. Hoping to get it merged by EOW.

**Comment by sakluger** [comment]
Got it, thanks, I've updated the title with the new hold issue, and am going to move the priority back to weekly.

**Comment by NikkiWines** [comment]
Sorry, meant to update this - the internal PR is on production now

**Comment by sakluger** [comment]
I asked for another QA retest in Slack.

**Comment by sakluger** [comment]
QA confirmed that this is not fixed yet 😢 

https://github.com/Expensify/App/assets/5997400/25f4faa7-4ca7-4616-a1fd-3d5950f3d6ee

Uploading -511377131899782377232243_web.MP4…

**Comment by sakluger** [comment]
Now that the other issue is on prod I've removed the hold. @Christinadobrzyn can you take this one back over?

@NikkiWines what do you think the next steps are now that we know the hold issue didn't fix it?

**Comment by NikkiWines** [comment]
I'll look into this tomorrow to see where it's going awry

**Comment by iwiznia** [comment]
I am not entirely sure, but https://github.com/Expensify/Web-Expensify/pull/40608/ is deployed and might have fixed this, can someone retest this to see if it is still reproducible?

**Comment by NikkiWines** [comment]
Looks like this is still reproducible. The total of the report is updated but the $4.50 expense remains on the report: 

![image](https://github.com/Expensify/App/assets/16747822/12ead2bf-7c6c-498c-a00e-57f7ea4b652e)
![image](https://github.com/Expensify/App/assets/16747822/71fa5d9a-c621-4dd0-b12f-48d39c79201c)
![image](https://github.com/Expensify/App/assets/16747822/3f35413d-e7a0-457f-bbd6-147c3eaaff51)

**Comment by greg-schroeder** [comment]
Making this a `Hot Pick` as Nikki is already tied to another What's Next issue for Wave 5

**Comment by yuwenmemon** [comment]
I'll take a look at this one!

**Comment by blimpich** [comment]
@yuwenmemon you're working on this correct? If so can we remove the hot pick label?

**Comment by yuwenmemon** [comment]
Yes sorry I keep forgetting to do that

**Comment by yuwenmemon** [comment]
Based on my testing, this is happening in real-time, and is working as expected. If no one has any objections I'm just gonna close this out 😄 

https://github.com/Expensify/App/assets/4741899/f483b8e7-c017-451b-b6e9-f5b848f3a768
